applications/OptimizationApplication/python_scripts/responses/linear_strain_energy_response_function.py

```diff
@@ -6,11 +6,13 @@
 from KratosMultiphysics.OptimizationApplication.responses.response_function import SupportedSensitivityFieldVariableTypes
 from KratosMultiphysics.OptimizationApplication.utilities.helper_utilities import ConvertCollectiveExpressionValueMapToModelPartValueMap
 
-def Factory(model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem) -> ResponseFunction:
-    return LinearStrainEnergyResponseFunction(model, parameters, optimization_problem)
+def Factory(name: str, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem) -> ResponseFunction:
+    return LinearStrainEnergyResponseFunction(name, model, parameters, optimization_problem)
 
 class LinearStrainEnergyResponseFunction(ResponseFunction):
-    def __init__(self, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+    def __init__(self, name: str, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        super().__init__(name)
+
         default_settings = Kratos.Parameters("""{
             "combined_output_model_part_name": "<RESPONSE_NAME>_combined_no_neighbours_response",
```

Comment:
This option is not self explaining, and I would not expose it to the user, what if there is no overlap between evaluated and sensitivity model part, from the implementation it looks like you combine the evaluated model parts.

Reply:
makes sense, and removed.